servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/MessageReceiver.java

```diff
@@ -762,17 +763,9 @@ public IMessage peek(long fromSequenceNumber) throws InterruptedException, Servi
 
     private void schedulePruningRequestResponseLockTokens() {
         // Run it every 1 hour
-        Timer.schedule(new Runnable() {
-            public void run() {
-                Instant systemTime = Instant.now();
-                Entry<UUID, Instant>[] copyOfEntries = (Entry<UUID, Instant>[]) MessageReceiver.this.requestResponseLockTokensToLockTimesMap.entrySet().toArray();
-                for (Entry<UUID, Instant> entry : copyOfEntries) {
-                    if (entry.getValue().isBefore(systemTime)) {
-                        // lock expired
-                        MessageReceiver.this.requestResponseLockTokensToLockTimesMap.remove(entry.getKey());
-                    }
-                }
-            }
+        Timer.schedule(() -> {
+            Instant systemTime = Instant.now();
+            MessageReceiver.this.requestResponseLockTokensToLockTimesMap.entrySet().removeIf(entry -> entry.getValue().isBefore(systemTime));
```

Comment:
>MessageReceiver.this.requestResponseLockTokensToLockTimesMap.entrySet().removeIf(entry -> entry.getValue().isBefore(systemTime)) [](start = 12, length = 128)

What are you trying to fix with this change? Downcast is flagged by your tool? Making a copy of the entry set is needed for thread safety. You removed that making a copy part , and running your predicate right on the hasmap itself. That's not what I Intended.